apiserver/application/application.go

```diff
@@ -888,6 +883,14 @@ func (api *API) AddRelation(args params.AddRelation) (params.AddRelationResults,
 		if relName != "" {
 			endpoints[i] = remoteApp.Name() + ":" + relName
 		}
+		isRemote = true
+	}
+	// If it's not a remote relation to another model then
+	// the user needs write access to the model.
+	if !isRemote {
```

Comment:
Actually, you are right, we still need to check for access on this model too. Oops. Fixed.